client/app/bundles/course/level/components/LevelsTable.tsx

```diff
@@ -0,0 +1,158 @@
+import {
+  Button,
+  IconButton,
+  Table,
+  TableBody,
+  TableCell,
+  TableFooter,
+  TableHead,
+  TableRow,
+  TextField,
+  Tooltip,
+  Typography,
+} from '@mui/material';
+
+import useTranslation from 'lib/hooks/useTranslation';
+
+const translations = defineMessages({
+  levelHeader: {
+    id: 'course.level.Level.levelHeader',
+    defaultMessage: 'Level',
+  },
+  thresholdHeader: {
+    id: 'course.level.Level.thresholdHeader',
+    defaultMessage: 'Threshold',
+  },
+  addNewLevel: {
+    id: 'course.level.Level.addNewLevel',
+    defaultMessage: 'Add New Level',
+  },
+  orderedIncorrectly: {
+    id: 'course.level.Level.orderedIncorrectly',
+    defaultMessage:
+      'Levels will be sorted automatically when saved regardless of their order here.',
+  },
+});
+
+interface LevelsTableProps {
+  levels: { level: number; deleted: boolean }[];
+  canManage: boolean;
+  isSaving: boolean;
+  onAddLevel: () => void;
+  onThresholdChange: (index: number, newThreshold: string) => void;
+  onDeleteLevel: (index: number) => void;
+  onUndoDelete: (index: number) => void;
+}
+
+const getClassName = (deleted: boolean, index: number): string => {
+  if (deleted) return 'bg-red-100'; // Red tint when marked for deletion
+  return index % 2 === 0 ? 'bg-white' : 'bg-gray-100';
+};
+
+const LevelsTable: FC<LevelsTableProps> = ({
+  levels,
+  canManage,
+  isSaving,
+  onAddLevel,
+  onThresholdChange,
+  onDeleteLevel,
+  onUndoDelete,
+}) => {
+  const { t } = useTranslation();
+  const isThresholdOrderedCorrectly = (index: number): boolean =>
+    index === 0 || levels[index].level > levels[index - 1].level;
+
+  return (
+    <Table>
+      <TableHead>
+        <TableRow>
+          <TableCell>
+            <FormattedMessage {...translations.levelHeader} />
+          </TableCell>
+          <TableCell>
+            <FormattedMessage {...translations.thresholdHeader} />
+          </TableCell>
+          <TableCell />
+        </TableRow>
+      </TableHead>
+
+      <TableBody>
+        {levels.slice(1).map((row, index) => (
+          <TableRow
+            // Index can be used as key as levels are not reordered
+            // eslint-disable-next-line react/no-array-index-key
+            key={`level-row-${index}`}
+            className={getClassName(row.deleted, index)}
+          >
+            <TableCell>
+              <Typography>{index + 1}</Typography>
+            </TableCell>
+
+            <TableCell>
+              <Tooltip
+                arrow
+                placement="top"
+                title={
+                  !isThresholdOrderedCorrectly(index + 1)
+                    ? t(translations.orderedIncorrectly)
+                    : ''
+                }
+              >
+                <TextField
+                  disabled={!canManage || isSaving || row.deleted}
+                  fullWidth
+                  onChange={(e) => onThresholdChange(index, e.target.value)}
+                  type="number"
```

Comment:
Can also consider reducing the width of this TextField. The current one that you made is quite spacious while we don't really normally use the space that huge for this matter

Reply:
Point 1: This behaviour prevents the user from entering values that are not numbers, so i think we should keep it. The zero also disappears once the user starts typing a number so i think its fine. Additionally the threshold is also stored as a number and it doesnt really makes sense to add a string type to it just to accommodate this as well.

Reply:
If you have inspected that `ExperiencePointsNumberField.tsx` and go into the page (which is under Experience Points tab in our Sidebar), you will notice that other than '-', we do not allow any non-numerical character to be inputted over there. You can just try modifying the behavior a bit to only allow numerical character for your Level Table.

And no, the zero won't disappear even after user starts typing a number (one scenario to be thought of would be to backspace the whole one of the levels, say level 1 of threshold `200`, then you will eventually see number 0 suddenly appear.. then you type the number you want to input, let's say `50`, then it will appear as `050`. Even though, after saving, that leading 0 will disappear, the UI is not good for user to see.

Reply:
One more thing, please put the placeholder to be what the level's threshold's initial value. In case user edited one of the level accidentally and want to revert back to what it's used to be, they can just erase the whole block and they might see what's the level threshold initially is

Again, please refer to that file for the complete behavior

Reply:
Done